File: python/app/adapter/driving/request_helper.py
from app.adapter.driving.not_login_exception import NotLoginException
from app.application.user_auth_service import UserAuthService
from app.application.user_dto import UserDto
from app.application.user_query_handler import UserQueryHandler
from app.ioc_container import ioc_container

user_auth_service = ioc_container.resolve(UserAuthService)  # type: ignore
user_query_handler = ioc_container.resolve(UserQueryHandler)  # type: ignore


def get_login_user_id(request) -> str:
    """获取登录用户Id，未登录返回空字符串"""
    access_token = request.headers.get("Access-Token", "")
    return user_auth_service.get_login_user_id(access_token)


def require_login_user_id(request) -> str:
    """获取登录用户Id，未登录抛出异常"""
    user_id = get_login_user_id(request)
    if user_id is None:
        raise NotLoginException()
    return user_id


def require_login_user_admin(request) -> UserDto:
    """获取登录管理员用户，失败抛异常"""
    user_id = get_login_user_id(request)
    user_dto = user_query_handler.query_by_id(user_id)
    if user_dto is None or not user_dto.is_admin:
        raise NotLoginException()
    return user_dto

File: java/src/main/java/com/example/ddd/adapter/driving/RequestHelper.java
package com.example.ddd.adapter.driving;

import com.example.ddd.application.UserAuthService;
import com.example.ddd.application.UserDto;
import com.example.ddd.application.UserQueryHandler;
import com.example.ddd.utility.JsonUtility;
import com.example.ddd.utility.SpringContextUtility;
import com.fasterxml.jackson.databind.JsonNode;
import io.micrometer.common.util.StringUtils;
import jakarta.servlet.http.HttpServletRequest;

/**
 * 请求 Helper
 */
public class RequestHelper {

    /**
     * 获取请求体JSON内容
     */
    public static JsonNode toJson(String requestBody) {
        try {
            return JsonUtility.deserialize(requestBody);
        }
        catch (Exception e) {
            throw new ControllerException("请求体不是合法的JSON格式");
        }
    }

    /**
     * 获取登录用户Id，未登录返回空字符串
     */
    public static String getLoginUserId(HttpServletRequest request) {
        String accessToken = request.getHeader("Access-Token");
        return SpringContextUtility.getBean(UserAuthService.class).getLoginUserId(accessToken);
    }

    /**
     * 获取登录用户Id，未登录抛异常
     */
    public static String requireLoginUserId(HttpServletRequest request) {
        String userId = getLoginUserId(request);
        if (StringUtils.isBlank(userId)) {
            throw new NotLoginException();
        }
        return userId;
    }

    /**
     * 获取登录管理员用户，失败抛异常
     */
    public static UserDto requireLoginUserAdmin(HttpServletRequest request) {
        String userId = requireLoginUserId(request);
        UserDto userDto = SpringContextUtility.getBean(UserQueryHandler.class).queryById(userId);
        if (userDto == null || !userDto.isAdmin()) {
            throw new NotLoginException();
        }
        return userDto;
    }
}
